### PR TITLE
fix(docker): exclude path from registry

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Fixes docker logic to exclude any path from the registry and add it to the repository instead.

## Context:

Related to #10135 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/docker/common.ts [hunk 1]
```diff
@@ -165,9 +165,14 @@ export function getRegistryRepository(
       if (!/^https?:\/\//.test(registryHost)) {
         registryHost = `https://${registryHost}`;
       }
+      let dockerRepository = lookupName.replace(registryEndingWithSlash, '');
+      const fullUrl = `${registryHost}/${dockerRepository}`;
+      const fullUrlParsed = new URL(fullUrl);
```
Comment (viceice) [inline lib/datasource/docker/common.ts:170]:
```suggestion
      const fullUrlParsed = parseUrl(fullUrl);
```

from `lib/url` ?

File: lib/datasource/docker/common.ts [hunk 2]
```diff
@@ -165,9 +165,14 @@ export function getRegistryRepository(
...
+      registryHost = fullUrlParsed.origin;
+      dockerRepository = fullUrl.replace(registryHost, '').substring(1);
```
Comment (viceice) [inline lib/datasource/docker/common.ts:172]: use `fullUrlParsed.pathname` ?

File: lib/datasource/docker/common.ts [hunk 3]
```diff
@@ -165,9 +169,14 @@ export function getRegistryRepository(
       if (!/^https?:\/\//.test(registryHost)) {
         registryHost = `https://${registryHost}`;
       }
+      let dockerRepository = lookupName.replace(registryEndingWithSlash, '');
+      const fullUrl = `${registryHost}/${dockerRepository}`;
+      const { origin, pathname } = parseUrl(fullUrl);
```
Comment (viceice) [inline lib/datasource/docker/common.ts:174]: Should we catch `null` returned by `parseUrl`? or simply buble null-ref up?

Comment (rarkins) [reply lib/datasource/docker/common.ts:174]: We should ideally have validated `registryUrls` earlier, but I'm not sure if we do.

**Review by viceice — APPROVED** [review]

**Pull request merged**